### DISCUSSION TITLE
vNext supported target frameworks

### DISCUSSION THREAD

**Discussion by mclift** [dotnet-state-machine/stateless#525] [General]
This is a call for feedback about supported target frameworks, prompted by PR #524 which was kindly raised by @pentp.

The PR focuses on LTS versions of the target frameworks, so it removes .NET 5.0 and replaces the Framework 4.x targets with Framework 4.6.2. As out-of-support Framework versions 4.0 and 4.5 are removed, it also removes .NET Standard 1.0, which is Microsoft's recommendation: https://learn.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-1-0#which-net-standard-version-to-target. So the proposed target frameworks are:

* .NET Standard 2.0
* .NET Framework 4.6.2
* .NET 6.0

This would be a candidate for a major version release of Stateless.

**Comment by leeoades** [comment]
I agree with this move. It is not unreasonable to expect consumers of stateless (or any library) to be using supported versions of the frameworks. After all, if Microsoft isn't supporting them, why should we?
